web/cobrands/fixmystreet/offline.js

```diff
@@ -0,0 +1,415 @@
+                savedForm = form[1];
+            }
+        });
+        if (savedForm) {
+            savedForm.replace(/\+/g, '%20').split('&').forEach(function(kv) {
+                kv = kv.split('=', 2);
+                if (kv[0] != 'save_inspected' && kv[0] != 'public_update' && kv[0] != 'save') {
+                    $('[name=' + kv[0] + ']').val(decodeURIComponent(kv[1]));
+                }
+            });
+        }
+
+        $('#report_inspect_form').submit(function() {
+            var data = $(this).serialize() + '&save=1&saved_at=' + Math.floor(+new Date() / 1000);
+            fixmystreet.offlineData.addForm(this.action, data);
+            location.href = '/my/planned?saved=1';
+            return false;
+        });
+        return true;
+    }
+
+    function replaceImages(selector) {
+        $(selector).each(function(i, img) {
+            if (img.src.indexOf('/photo/') > -1) {
+                var dataImg = localStorage.getItem(img.src);
+                if (dataImg) {
+                    img.src = dataImg;
+                }
+            }
+        });
+    }
+
+    return {
+        replaceImages: replaceImages,
+        showReportFromCache: showReportFromCache,
+        removeReports: removeReports,
+        updateCachedReports: updateCachedReports
+    };
+
+})();
+
+if ($('#offline_list').length) {
+    // We are OFFLINE
+    var success = false;
+    if (location.pathname.indexOf('/report') === 0) {
+        success = fixmystreet.offline.showReportFromCache(location.pathname);
+    }
+    if (!success) {
+        var my_planned = localStorage.getItem('/my/planned');
+        var html = my_planned || 'Nothing cached offline - <a href="/my/planned">View your shortlist online to cache</a>';
+        $('#offline_list').html(html);
+        if (location.search.indexOf('saved=1') > 0) {
+            $('#offline_list').before('<p class="form-success">Your form has been saved offline for submission when back online.</p>');
+        }
+        fixmystreet.offline.replaceImages('#offline_list img');
+        var offlineForms = fixmystreet.offlineData.getForms();
+        var savedForms = {};
+        offlineForms.forEach(function(form) {
+            savedForms[form[0]] = 1;
+        });
+        $('#offline_list a').each(function(i, a) {
+            if (savedForms[a.href]) {
+                $(this).find('h3').prepend('<em>Form data saved</em> ');
+            }
+        });
+        if (my_planned || offlineForms.length) {
+            $('#offline_clear').html('<button id="js-clear-localStorage">Clear offline data</button>');
+            $('#js-clear-localStorage').click(function() {
+                fixmystreet.offline.removeReports(fixmystreet.offlineData.getCachedUrls());
+                fixmystreet.offlineData.clearForms();
+                localStorage.removeItem('/my/planned');
+                alert('Offline data cleared');
+            });
+        }
+    }
+    fixmystreet.offlineBanner.make(true);
+} else {
```

Comment:
This doesn't seem to limit the offline preparation to logged-in inspectors, so everyone's browser will add the iframe on every page load.